crates/rust-analyzer/src/config.rs

```diff
@@ -917,6 +917,9 @@ impl Config {
             snippets: self.snippets.clone(),
         }
     }
+    pub fn snippet_cap(&self) -> bool {
+        self.experimental("snippetTextEdit")
```

Comment:
I think this capability might only indicate support for snippets in code actions / assists, so we might need a new capability for this.

Reply:
Do you have a suggestion for the new capability?  This PR extends current capability by updating /docs/dev/lsp-extensions.md

Reply:
@jonas-schievink, I've pushed a new commit to the PR addressing one of your concerns.  It now does nothing when a "closing angle bracket is already there".  

Regarding your other concern, I still think it's a good idea to extend the meaning of the `snippetTextEdit` client capability as the current PR does.  However, I've come up with the following alternatives that would somewhat complicate the implementation.  The `lsp-extensions.md` currently has this:
```
{ "snippetTextEdit": boolean }
```
It could be modified to this:
```
{ "snippetTextEdit": boolean | SnippetCapabilities; }
export interface SnippetCapabilities {
  /** 
   * Whether client supports `SnippetTextEdit`s in `WorkspaceEdit`s returned from `codeAction` requests.
   */
  codeAction: boolean;

  /**
   *  Whether client supports `SnippetTextEdit`s in replies to `onTypeFormating` requests.
   */
  onTypeFormatting?:  boolean;
}
```
or even 
```
  [the same as above plus:]
  /**
   *  Whether client can process `SnippetTextEdit` in all the places where the specification otherwise requires `TextEdit`.
   */
  everywhere?: boolean;
}
```
What do you think?

Reply:
Would this cause problems for clients if we extended the meaning of the capability here? That is what would happen to clients that do not expect onTyping to return a meaningful snippet edit for `<`.

Reply:
I assume there is a generic apply-text-edits subroutine/function in the client that checks whether a TextEdit has a insertTextFormat property and acts accordingly.  It would be counter-intuitive to me to implement two functions, one for TextEdits and one for SnippetTextEdits, because the latter can be a simple extension to the former.  

However, if you think my assumption is wrong, I proposed protocol extensions above to explicitly handle this situation.   If you decide which one is preferable, I can try to implement it.